src/earn/saga.ts

```diff
@@ -117,61 +118,70 @@ export function* depositSubmitSaga(action: PayloadAction<DepositInfo>) {
-        ): BaseStandbyTransaction => {
-          return {
-            context: newTransactionContext(TAG, 'Earn/Approve'),
-            __typename: 'TokenApproval',
-            networkId,
-            type: TokenTransactionTypeV2.Approval,
-            transactionHash,
-            tokenId,
-            approvedAmount,
-            feeCurrencyId,
+    if (preparedTransactions.length <= 2) {
+      // if there are 1 or 2 transactions, its an approve (optional) and deposit
+      if (preparedTransactions.length > 1 && preparedTransactions[0].data) {
+        const { functionName, args } = decodeFunctionData({
+          abi: erc20Abi,
+          data: preparedTransactions[0].data,
+        })
+        if (
+          functionName === 'approve' &&
+          preparedTransactions[0].to === tokenInfo.address &&
+          args
+        ) {
+          Logger.debug(`${TAG}/depositSubmitSaga`, 'First transaction is an approval transaction')
+          const approvedAmountInSmallestUnit = args[1] as bigint
+          const approvedAmount = new BigNumber(approvedAmountInSmallestUnit.toString())
+            .shiftedBy(-tokenInfo.decimals)
+            .toString()
+
+          const createApprovalStandbyTx = (
+            transactionHash: string,
+            feeCurrencyId?: string
+          ): BaseStandbyTransaction => {
+            return {
+              context: newTransactionContext(TAG, 'Earn/Approve'),
+              __typename: 'TokenApproval',
+              networkId,
+              type: TokenTransactionTypeV2.Approval,
+              transactionHash,
+              tokenId,
+              approvedAmount,
+              feeCurrencyId,
+            }
           }
+          createDepositStandbyTxHandlers.push(createApprovalStandbyTx)
         }
-        createDepositStandbyTxHandlers.push(createApprovalStandbyTx)
       }
-    }
 
-    // this is a placeholder for now, should be updated to a "deposit" transaction type
-    const createDepositStandbyTx = (
-      transactionHash: string,
-      feeCurrencyId?: string
-    ): BaseStandbyTransaction => {
-      return {
-        context: newTransactionContext(TAG, 'Earn/Deposit'),
-        __typename: 'EarnDeposit',
-        networkId,
-        type: TokenTransactionTypeV2.EarnDeposit,
-        inAmount: {
-          value: amount,
-          tokenId: networkConfig.aaveArbUsdcTokenId,
-        },
-        outAmount: {
-          value: amount,
-          tokenId,
-        },
-        providerId: 'aave-v3',
-        transactionHash,
-        feeCurrencyId,
+      const createDepositStandbyTx = (
+        transactionHash: string,
+        feeCurrencyId?: string
+      ): BaseStandbyTransaction => {
+        return {
+          context: newTransactionContext(TAG, 'Earn/Deposit'),
+          __typename: 'EarnDeposit',
+          networkId,
+          type: TokenTransactionTypeV2.EarnDeposit,
+          inAmount: {
+            value: amount,
+            tokenId: networkConfig.aaveArbUsdcTokenId,
```

Comment:
Should we use `earnPosition.withdrawTokenId` instead?
Though it's possible it could be wrong in the future, with new type of pools.

Reply:
oh good catch, I think we can also use the positionId? maybe that's safer that withdrawTokenId?

Reply:
Ah yes but the positionId only works in some cases, if the position is an app-token.
For instance it wouldn’t work if we had an earn position for staking celo natively which is modeled using a contract-position as there’s not another erc20 token the user gets back representing it.

withdrawTokenId seems safer then.

Reply:
In the future I wonder if we should just show this differently in the feed so we don’t need in and out which don’t apply in all cases.